### PR TITLE
docs: add release notes for 0.34.0

### PR DESCRIPTION
**Pull request by keita-determined**
Add release notes for 0.34.0.

### CODE REVIEW

File: docs/release-notes.rst [hunk 1]
```diff
@@ -6,6 +6,98 @@
...
 ###############
 
+**************
+ Version 0.34
+**************
+
+Version 0.34.0
+==============
+
+**Release Date:** June 24, 2024
```
Comment (keita-determined) [inline docs/release-notes.rst:16]: # TODO: fix the release date before merging it!

File: docs/release-notes.rst [hunk 2]
```diff
@@ -6,6 +6,98 @@
...
+
+**Breaking Changes**
+
+-  Images: The default environment includes images that support PyTorch. TensorFlow users must
+   configure their experiments to target our non-default TensorFlow images. Details on this process
+   can be found at :ref:`set-environment-images`
```
Comment (tara-det-ai) [inline docs/release-notes.rst:22]: -  Images: The default environment includes images that support PyTorch. Therefore, TensorFlow users must
   configure their experiments to target our non-default TensorFlow images. Details on this process
   can be found at :ref:`set-environment-images`

File: docs/release-notes.rst [hunk 3]
```diff
@@ -6,6 +6,98 @@
...
+
+-  Images: Our new default images are based on Nvidia NGC. While we provide a recommended NGC
+   version, users have the flexibility to build their own images using any NGC version that meets
+   their specific requirements. For more information, visit :ref:`ngc-version`
+
```
Comment (tara-det-ai) [inline docs/release-notes.rst:27]: -  Images: Our new default images are based on Nvidia NGC. While we provide a recommended NGC
   version, users can build their own images using any NGC version that meets
   their specific requirements. For more information, visit :ref:`ngc-version`

File: docs/release-notes.rst [hunk 4]
```diff
@@ -6,6 +6,98 @@
...
+   deployed on Kubernetes with multiple resource pools defined in terms of node selectors and/or
+   affinities.
+
+**Bug Fixes**
+
+-  Kubernetes: Fix an issue where where jobs would remain in "QUEUED" state until all pods were
+   running. Jobs will now correctly show as "SCHEDULED" once all pods have been assigned to nodes.
+
+**ZZZ put these into an appropriate section!**
+
```
Comment (tara-det-ai) [inline docs/release-notes.rst:85]: :-)

**Review by tara-det-ai — APPROVED** [review]
suggested edits

File: docs/release-notes.rst (hunk 2 above)
Comment (tara-det-ai) [inline docs/release-notes.rst:22]:
```suggestion
-  Images: The default environment includes images that support PyTorch. Therefore, TensorFlow users must configure their experiments to target our non-default TensorFlow images. Details on this process can be found at :ref:`set-environment-images`
```

File: docs/release-notes.rst [hunk 5]
```diff
@@ -6,6 +6,98 @@
...
+**************
+
+Version 0.34.0
+==============
+
+**Release Date:** June 24, 2024
+
+**Breaking Changes**
+
+-  Images: The default environment includes images that support PyTorch. TensorFlow users must
```
Comment (tara-det-ai) [inline docs/release-notes.rst:20]:
```suggestion
-  Images: The default environment includes images that support PyTorch. Therefore, TensorFlow users must
```

File: docs/release-notes.rst [hunk 6]
```diff
@@ -6,6 +6,98 @@
...
+   configure their experiments to target our non-default TensorFlow images. Details on this process
+   can be found at :ref:`set-environment-images`
+
+-  Images: Our new default images are based on Nvidia NGC. While we provide a recommended NGC
+   version, users have the flexibility to build their own images using any NGC version that meets
```
Comment (tara-det-ai) [inline docs/release-notes.rst:25]:
```suggestion
version, users can build their own images using any NGC version that meets
```

**Review by tara-det-ai — APPROVED** [review]
suggested edits

File: docs/release-notes.rst (hunk 4 above)
Comment (keita-determined) [reply docs/release-notes.rst:85]: its ok to leave `ZZZ`?

File: docs/release-notes.rst [hunk 7]
```diff
@@ -6,6 +6,98 @@
...
+**************
+
+Version 0.34.0
+==============
+
+**Release Date:** June 24, 2024
+
+**Breaking Changes**
+
+-  Images: The default environment includes images that support PyTorch. Therefore, TensorFlow users
```
Comment (tara-det-ai) [inline docs/release-notes.rst:20]:
```suggestion
- Images: The default environment includes images that support PyTorch. Therefore, TensorFlow users.
```

File: docs/release-notes.rst [hunk 8]
```diff
@@ -6,6 +6,98 @@
...
+   must configure their experiments to target our non-default TensorFlow images. Details on this
+   process can be found at :ref:`set-environment-images`
```
Comment (tara-det-ai) [inline docs/release-notes.rst:22]:
```suggestion
must configure their experiments to target our non-default TensorFlow images. Details on this
   process can be found at :ref:`set-environment-images`.
```

**Comment by ioga** [comment]
can we please also cherry-pick and include the note from https://github.com/determined-ai/determined/pull/9571

**Comment by keita-determined** [comment]
> can we please also cherry-pick and include the note from #9571

@ioga cherry-picked

File: docs/release-notes.rst (hunk 4 above)
Comment (tara-det-ai) [reply docs/release-notes.rst:87]: we should remove it. i thought it was a reminder for someone? maybe not.

Comment (keita-determined) [reply docs/release-notes.rst]: fixed

**Pull request merged**